docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+# Devfile Versioning and Release Process
+This design document outlines the proposed versioning and release process for the Devfile spec. 
+
+This document summarizes parts from he Devfile API technical meeting slides presented by @davidfestal back in October, and I recommend having a read through of it: https://docs.google.com/presentation/d/1ohM1HzPB59a3ajvB7rVWJkKONLBAuT0mb5P20_A6vLs/edit#slide=id.g8fc722bef9_1_8
+
+## Versioning
+
+The following sections outline how we version the Devfile Kubernetes API, as well as the JSON schema generated from the API.
+
+### Kubernetes API
+The Devfile Kubernetes API (defined in https://github.com/devfile/api/) is the single source of truth for the Devfile spec. It’s defined under `pkg/apis/` and contains the Go structs that are used by the devworkspace operator and devfile library. It's also where the Devfile JSON schema is generated from. 
+
+**Versioning Scheme**: [Kubernetes](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) (e.g. v1alpha1, v1beta2, v1, v2, etc)
+
+**Update Process**: 
+
+   1) Add a new folder in the source repository (`pkg/apis/workspaces` in `devfile/api`)
+   2) Add a schema and version in the CRD manifests
+   3) Generate the JSON schema from the API. New JSON schema will be located under `schemas/latest`.
+   4) Update the devworkspace operator and devfile library to consume the Go structs in the new API version, as needed.
+
+**When to Update?**  As incrementing the Kubernetes API version for Devfile is a relatively heavy process, and affects the library, only update the K8s API version when absolutelyneeded 
+   - for **big** changes or backwards incompatible changes.
```

Comment:
We may want to define "backwards incompatible" more precisely -- technically speaking, almost all changes to Go structs are backwards incompatible (in a not-serious way for our cases). Perhaps some language referring to json schema compatibility (adding fields is fine, removing them is not) would make this clearer. My metric is "can I apply the same resource I used to without it failing k8s validation?"

Reply:
Yeah, I think that sounds reasonable 👍

Reply:
Updated. Added a couple bullet points on backwards-incompatible changes (and when a non-breaking change may still require a version bump)